score-xml/src/main/java/io/cloudslang/content/xml/services/RemoveService.java

```diff
@@ -24,22 +24,25 @@
         Map<String, String> result = new HashMap<>();
 
         try {
-            Document doc = XmlUtils.parseXML(commonInputs.getXmlDocument(), commonInputs.getSecureProcessing());
-            NamespaceContext context = XmlUtils.createNamespaceContext(commonInputs.getXmlDocument());
+            Document doc = XmlUtils.getDocument(commonInputs);
+            NamespaceContext context = XmlUtils.getNamespaceContext(commonInputs, doc);
             NodeList nodeList = XmlUtils.evaluateXPathQuery(doc, context, commonInputs.getXPathQuery());
 
             XmlUtils.validateNodeList(nodeList);
 
             removeFromNodeList(nodeList, customInputs.getAttributeName());
-            ResultUtils.populateSuccessResult(result, Constants.SuccessMessages.REMOVE_SUCCESS,
-                    XmlUtils.nodeToString(doc));
+            ResultUtils.populateValueResult(result, Constants.SUCCESS, Constants.SuccessMessages.REMOVE_SUCCESS,
+                    XmlUtils.nodeToString(doc), Constants.ReturnCodes.SUCCESS);
 
         } catch (XPathExpressionException e) {
-            ResultUtils.populateFailureResult(result, Constants.ErrorMessages.XPATH_PARSING_ERROR + e.getMessage());
```

Comment:
@tethryus Should be refactored to use some kind of generic static method somewhere in OutputsUtil class for e.g.